From 718178405f3ff6d7e2f64544edcb21429289a111 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Wed, 12 Jan 2022 11:42:19 +0800
Subject: [PATCH 5/5] HACK: Bypass auth APIs by default

There're hacks in Rockchip's BSP kernels to ignore drm auth, so we can
bypass the related drm auth APIs here.

Bypass these auth APIs:
drmGetBusid, drmAuthMagic, drmSetInterfaceVersion, drmSetMaster,
drmDropMaster.

Set these environments to enable:
"DRM_GET_BUSID", "DRM_AUTH_MAGIC", "DRM_SET_VERSION", "DRM_SET_MASTER",
"DRM_DROP_MASTER".

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 xf86drm.c | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/xf86drm.c b/xf86drm.c
index 7f67a63..ae3f4bc 100644
--- a/xf86drm.c
+++ b/xf86drm.c
@@ -1514,6 +1514,9 @@ drm_public char *drmGetBusid(int fd)
 {
     drm_unique_t u;
 
+    if (!getenv("DRM_GET_BUSID"))
+        return strdup("");
+
     memclear(u);
 
     if (drmIoctl(fd, DRM_IOCTL_GET_UNIQUE, &u))
@@ -1572,6 +1575,9 @@ drm_public int drmAuthMagic(int fd, drm_magic_t magic)
 {
     drm_auth_t auth;
 
+    if (!getenv("DRM_AUTH_MAGIC"))
+        return 0;
+
     memclear(auth);
     auth.magic = magic;
     if (drmIoctl(fd, DRM_IOCTL_AUTH_MAGIC, &auth))
@@ -3070,6 +3076,9 @@ drm_public int drmSetInterfaceVersion(int fd, drmSetVersion *version)
     int retcode = 0;
     drm_set_version_t sv;
 
+    if (!getenv("DRM_SET_VERSION"))
+        return 0;
+
     memclear(sv);
     sv.drm_di_major = version->drm_di_major;
     sv.drm_di_minor = version->drm_di_minor;
@@ -3268,11 +3277,17 @@ drm_public void drmCloseOnce(int fd)
 
 drm_public int drmSetMaster(int fd)
 {
-        return drmIoctl(fd, DRM_IOCTL_SET_MASTER, NULL);
+	if (!getenv("DRM_SET_MASTER"))
+		return 0;
+
+	return drmIoctl(fd, DRM_IOCTL_SET_MASTER, NULL);
 }
 
 drm_public int drmDropMaster(int fd)
 {
+	if (!getenv("DRM_DROP_MASTER"))
+		return 0;
+
         return drmIoctl(fd, DRM_IOCTL_DROP_MASTER, NULL);
 }
 
-- 
2.20.1

